### PR TITLE
Rewrite launcher catalog pr command

### PR DESCRIPTION
**Pull request by gytis**
This does not fix actual PR creation issue yet. However, it introduced abstractions on GitHub, Updater as well as extracts the hardcoded values to the configuration.
It also fixes how the example tags are found.

https://github.com/snowdrop/snowdrop-automation-client/issues/84
https://github.com/snowdrop/snowdrop-automation-client/issues/73

### CODE REVIEW

**Comment by gytis** [comment]
I'll just change the indentation back to 2 spaces, since it's the dominant one around the project.

File: lib/commands/CreateLauncherPR.ts [hunk 1]
```diff
@@ -7,89 +7,109 @@ import {
...
-import {CommandHandler} from "@atomist/automation-client/lib/decorators";
-import {HandleCommand} from "@atomist/automation-client/lib/HandleCommand";
-import {relevantRepos} from "@atomist/automation-client/lib/operations/common/repoUtils";
-import {editOne} from "@atomist/automation-client/lib/operations/edit/editAll";
-import {BranchCommit} from "@atomist/automation-client/lib/operations/edit/editModes";
-import {allReposInTeam} from "@atomist/sdm";
-import {DefaultRepoRefResolver} from "@atomist/sdm-core";
-import {BOOSTER_CATALOG_REPO, SNOWDROP_ORG} from "../constants";
-import {updateLauncherCatalog} from "../support/catalog/updateLauncherCatalog";
-import {DefaultLatestTagRetriever} from "../support/github/boosterUtils";
```
Comment (jacobdotcosta) [inline lib/commands/CreateLauncherPR.ts:23]: Does removing the `LatestTagRetriever` has to do with the automation client calculating wrong the latest version? #73

File: config/default.json [hunk 2]
```diff
@@ -6,5 +6,19 @@
         "name": "Libor"
```
[the remark this answers is not in the document]
Comment (gytis) [reply config/default.json:6]: We could, but I'd suggest to do it as a separate change from this PR.

File: lib/commands/CreateLauncherPR.ts (hunk 1 above)
Comment (gytis) [reply lib/commands/CreateLauncherPR.ts:23]: Yes. I've extracted the logic to a separate place and started using an interface for abstraction https://github.com/snowdrop/snowdrop-automation-client/pull/85/files#diff-cb9f91a06d665c423e70ab3c87133645R1.
Also, `LatestTagRetriever` used to get the tags from github + filter them. I've split this logic and now tag finder only filters an array that is given to it. This way it's easier to test it.

**Comment by jacobdotcosta** [comment]
LGTM

**Comment by gytis** [comment]
That's fare enough :) I'm just trying to keep people in the loop.

**Comment by gytis** [comment]
The main idea is to start abstracting the logic that interacts with GitHub directly under an interface rather than a collection of loose methods. I think this way we might be able to make other parts of the code a little bit cleaner and more testable.

**Pull request merged**